pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,625 @@
+}
+
+message CreateContainerResponse {
+    // The id of the created container
+    optional string container_id = 1;
+}
+
+message StartContainerRequest {
+    // The id of the container
+    optional string container_id = 1;
+}
+
+message StartContainerResponse {}
+
+message StopContainerRequest {
+    // The id of the container
+    optional string container_id = 1;
+    // Timeout in seconds to stop the container
+    optional int64 timeout = 2;
+}
+
+message StopContainerResponse {}
+
+message RemoveContainerRequest {
+    // The id of the container
+    optional string container_id = 1;
+}
+
+message RemoveContainerResponse {}
+
+enum ContainerState {
+    CREATED = 0;
+    RUNNING = 1;
+    EXITED  = 2;
+    UNKNOWN = 3;
+}
+
+// ContainerFilter is used to filter containers.
+message ContainerFilter {
+    // Name of the container.
+    optional string name = 1;
+    // ID of the container.
+    optional string id = 2;
+    // State of the contianer.
+    optional ContainerState state = 3;
+    // The id of the pod sandbox
+    optional string pod_sandbox_id = 4;
+    // LabelSelector to select matches.
+    // Only api.MatchLabels is supported for now and the requirements
+    // are ANDed. MatchExpressions is not supported yet.
+    map<string, string> label_selector = 5;
+}
+
+message ListContainersRequest {
+    optional ContainerFilter filter = 1;
+}
+
+// Container provides the runtime information for a container, such as ID, hash,
+// state of the container.
+message Container {
+    // The ID of the container, used by the container runtime to identify
+    // a container.
+    optional string id = 1;
+    // The name of the container
+    optional string name = 2;
+    // Reference to the image in use. For most runtimes, this should be an
+    // image ID.
+    optional string image_ref = 3;
+    // State is the state of the container.
+    optional ContainerState state = 4;
+    // Labels are key value pairs that may be used to scope and select individual resources.
+    map<string, string> labels = 5;
+}
+
+message ListContainersResponse {
+    // List of containers
+    repeated Container containers = 1;
+}
+
+message ContainerStatusRequest {
+    // The id of the container
+    optional string container_id = 1;
+}
+
+// ContainerStatus represents the status of a container.
+message ContainerStatus {
+    // ID of the container.
+    optional string id = 1;
+    // Name of the container.
+    optional string name = 2;
+    // Status of the container.
+    optional ContainerState state = 3;
+    // Creation time of the container.
+    optional int64 created_at = 4;
+    // Start time of the container.
+    optional int64 started_at = 5;
+    // Finish time of the container.
+    optional int64 finished_at = 6;
+    // Exit code of the container.
+    optional int32 exit_code = 7;
+    // Reference to the image in use. For most runtimes, this should be an
+    // image ID
+    optional string image_ref = 8; 
```

Comment:
@feiskyer I am pretty sure the container status returned by `kubectl get pod -o  yaml` contains the image name it gets from the runtime.GetPodStatus(UID). So I still think we want to have a image name here, (or we populate the image with what's in the pod's spec, but the GetPodStatus() only takes a UID, so it requires kubelet to do additional jobs) cc @yujuhong